testing/citest/spinnaker_testing/azure_scenario_support.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2017 Microsoft Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Azure platform and test support for SpinnakerTestScenario."""
+
+import citest.azure_testing as az
+from spinnaker_testing.base_scenario_support import BaseScenarioPlatformSupport
+
+
+class AzureScenarioSupport(BaseScenarioPlatformSupport):
+  """Provides SpinnakerScenarioSupport for Azure."""
+
+  @classmethod
+  def add_commandline_parameters(cls, scenario_class, builder, defaults):
+    """Implements BaseScenarioPlatformSupport interface.
+
+    Args:
+      scenario_class: [class spinnaker_testing.SpinnakerTestScenario]
+      builder: [citest.base.ConfigBindingsBuilder]
+      defaults: [dict] Default binding value overrides.
+         This is used to initialize the default commandline parameters.
+    """
+    #
+    # Operation Parameters
+    #
+
+    # pylint: disable=line-too-long
+    # builder.add_argument(
+    #     '--azure_subscription_name', 
+    #     dest='spinnaker_azure_account',
+    #     help='The name of your subscriptoin.')
+    builder.add_argument(
+        '--azure_subscription_id', 
+        dest='spinnaker_azure_subscription_id',
+        help='The subscription id of your subscriptoin.')
+    builder.add_argument(
+        '--azure_tenant_id', 
+        dest='spinnaker_azure_tenant_id',
+        help='The AAD tenant id.')
+    builder.add_argument(
+        '--azure_client_id', 
+        dest='spinnaker_azure_client_id',
+        help='The service principal id.')
+    builder.add_argument(
+        '--azure_app_key', 
+        dest='spinnaker_azure_app_key',
+        help='The service principal secret.')
```

Comment:
I still think you should default the app key to an environment variable.
Passing the other parameters on the commandline is fine, but you wouldnt want to handle the credentials the same way so it just creates more work. You have a scheme where you get them all the same way from the deployment, but that wont work in general (e.g. when the deployment isnt azure, which wont always be the case). It isnt extra work because you dont have to use it, in which case you are back to where you already are here.